service/collector/orphan_configmap.go

```diff
@@ -0,0 +1,112 @@
+package collector
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/giantswarm/apiextensions/pkg/clientset/versioned"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"github.com/prometheus/client_golang/prometheus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/giantswarm/chart-operator/pkg/label"
+	"github.com/giantswarm/chart-operator/service/controller/chart/v1/key"
+)
+
+var (
+	orphanConfigMapDesc *prometheus.Desc = prometheus.NewDesc(
+		prometheus.BuildFQName(Namespace, "configmap", "orphan"),
+		"Configmaps without a chart CR.",
+		[]string{},
+		nil,
+	)
+)
+
+type OrphanConfigMapConfig struct {
+	G8sClient versioned.Interface
+	K8sClient kubernetes.Interface
+	Logger    micrologger.Logger
+}
+
+type OrphanConfigMap struct {
+	g8sClient versioned.Interface
+	k8sClient kubernetes.Interface
+	logger    micrologger.Logger
+}
+
+func NewOrphanConfigMap(config OrphanConfigMapConfig) (*OrphanConfigMap, error) {
+	if config.G8sClient == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.G8sClient must not be empty", config)
+	}
+	if config.K8sClient == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.K8sClient must not be empty", config)
+	}
+	if config.Logger == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Logger must not be empty", config)
+	}
+
+	oc := &OrphanConfigMap{
+		g8sClient: config.G8sClient,
+		k8sClient: config.K8sClient,
+		logger:    config.Logger,
+	}
+
+	return oc, nil
+}
+
+func (oc *OrphanConfigMap) Collect(ch chan<- prometheus.Metric) error {
+	oc.logger.Log("level", "debug", "message", "collecting metrics for orphan configmaps")
+
+	charts, err := oc.g8sClient.ApplicationV1alpha1().Charts("").List(metav1.ListOptions{})
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	desiredConfigMaps := make(map[string]bool)
```

Comment:
I'd prefer to make it 2 string array instead of constructing custom key. It's also more correct in terms of (unlikely) collisions. See https://play.golang.org/p/QMyCuNv0EO0.

```suggestion
	desiredConfigMaps := make(map[[2]string]bool)
```

Reply:
Thanks that is neat and more correct. Done.